chisel-tools/cmd/sdf/cmd_install.go

```diff
@@ -0,0 +1,342 @@
+// tasks and a channel to send errors to.
+func worker(ctx context.Context, tasks <-chan *task, errs chan<- error) {
+	// We are using an independent cache directory for chisel in each worker.
+	// The reason is tricky to detect. When creating files in cache, Chisel
+	// temporary saves a file as "<digest>.tmp" in the cache directory.[^1]
+	// It later renames the file to "<digest>" within the same directory.[^2]
+	// This is OK, if two chisel operations are sequential and/or do not try to
+	// "create" the cache at the same time. But on concurrent operations, the
+	// second process fails to rename the temporary file and eventually fails.
+	//
+	//   error: cannot fetch from archive: rename
+	//   /home/runner/.cache/chisel/sha256/6b8cc68643d18250ab297c4f6d427a8778b1d1534e1a3033fff0f221fa20a419.tmp
+	//   /home/runner/.cache/chisel/sha256/6b8cc68643d18250ab297c4f6d427a8778b1d1534e1a3033fff0f221fa20a419:
+	//   no such file or directory
+	//
+	// [^1]: https://github.com/canonical/chisel/blob/main/internal/cache/cache.go#L112
+	// [^2]: https://github.com/canonical/chisel/blob/main/internal/cache/cache.go#L80
+	cacheDir, err := os.MkdirTemp("", "")
+	if err != nil {
+		errs <- fmt.Errorf("cannot create temporary directory: %w", err)
+		return
+	}
+
+	do := func(task *task) {
+		name := strings.Join(task.slices, " ")
+		log.Printf("Installing %s...", name)
+
+		dir, err := os.MkdirTemp("", "")
+		if err != nil {
+			errs <- fmt.Errorf("cannot create temporary directory: %w", err)
+			return
+		}
+		defer os.RemoveAll(dir)
+
+		args := append(task.args, "--root", dir)
+		args = append(args, task.slices...)
+
+		cmd := exec.CommandContext(ctx, "chisel", args...)
+		cmd.Env = os.Environ()
+		cmd.Env = append(cmd.Env, "XDG_CACHE_HOME="+cacheDir)
+
+		if out, err := cmd.CombinedOutput(); err != nil {
+			if e, ok := err.(*exec.ExitError); ok && e.ProcessState.ExitCode() != -1 {
+				err = fmt.Errorf("%c Failed to install %s: %w", cross, name, err)
+				log.Printf("%s\n%s", err, out)
+			}
+			errs <- err
+		} else {
+			log.Printf("%c Installed %s", tick, name)
+		}
+	}
+
+loop:
+	for {
+		select {
+		case <-ctx.Done():
+			break loop // Context cancelled. Quit.
+		case task, ok := <-tasks:
+			if !ok {
+				break loop // No more tasks. Quit.
+			}
+			do(task)
+		}
+	}
+}
+
+// Query the archives for package existence, using the chisel.yaml
+// configurations.
+func (c *cmdInstall) queryArchive(slices []*chisel.Slice) (map[string]*rmadison.Result, error) {
+	p := filepath.Join(c.Release, "chisel.yaml")
+	cfg, err := chisel.ParseConfig(p)
+	if err != nil {
+		return nil, fmt.Errorf("cannot parse chisel.yaml: %w", err)
+	}
+	ubuntu, ok := cfg.Archives["ubuntu"]
+	if !ok {
+		return nil, fmt.Errorf("no 'ubuntu' archive in chisel.yaml")
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 20*time.Second)
+	defer cancel()
+
+	var pkgs []string
+	exists := make(map[string]struct{})
```

Comment:
```suggestion
	packageSeen := make(map[string]struct{})
```

And please add a comment here to indicate the purpose of using an empty struct.